### ISSUE TITLE
Last.fm Integration

### ISSUE THREAD

**Issue by huberf** [stevenleeg/geemusic#46]
I use the [last.fm](https://www.last.fm) service to keep a record of every song I play and this could be a valuable addition to the Alexa skill for users of this site. I am currently working on an integration on my local fork, and it would only rely on the requests library to carry out this functionality. Currently, I plan to have the integration be activated by the inclusion of an environment variable so people who don't use the service have to take no action to keep it disabled. Does this sound like a useful addition to the GeeMusic skill? I want to ensure the code base doesn't become cluttered with functionality, so wanted to ensure this would be a worthwhile addition.

It would make one request to the [last.fm API](http://www.last.fm/api) to "scrobble" the currently playing song, and so the code would only be activated when a new song starts streaming.

**Comment by fergyfresh** [comment]
This is going to sound a bit naïve, but could we trigger the 'scrobbling' outside of this skill and do it based on what is currently playing on Alexa so that way the last.fm skill can integrate with the Spotify skill?

**Comment by huberf** [comment]
I've gone ahead and started the code in a separate [GitHub repo](https://github.com/huberf/lastfm-scrobbler), and will open a PR once it is complete. I'll leave this issue open for now, in case someone else is interested in chiming in.

**Comment by fergyfresh** [comment]
https://github.com/pylast/pylast would that do what you want? The documentation is kind of lacking, but I just found it and wanted to push out the info to you before you got too far down the rabbit hole.

**Comment by Danondso** [comment]
Hmm what about a standalone scrobbling skill for Alexa? No clue about the work involved with that but it would be nice to have a scrobbler independent of service.

**Comment by huberf** [comment]
@fergyfresh That library isn't lightweight enough for what I want. The last.fm API is dead simple (a single POST request will scrobble a song), so writing my own shouldn't be very hard.

@Danondso I don't think a skill can get access to the APIs needed to acquire the currently playing song, but I'll look into it and see if it could be done. That would definitely be optimal, and if it is possible, I'll definitely build a skill for it.

**Comment by arimk** [comment]
Count me in, also using lastFM on all of my devices, just missing the Echo now !

**Comment by huberf** [comment]
I likely would have finished the entire system by now if the Last.fm API dashboard wasn't down. No matter what I do (using different devices, different URL endpoints) it always returns a 500 error when I try to acces the place where my API keys are controlled from. I have the code fully working except for this, and will update you all when it resolved. I haven't been able to get a response from support.

**Comment by huberf** [comment]
The API wrapper should be complete now and seems to function correctly. All the code can be accessed, tested, and contributed to at https://github.com/huberf/lastfm-scrobbler

Now, I just need to work on correctly integrating this into the GeeMusic repo.

**Comment by huberf** [comment]
I now have Last.fm support fully working. You can view it in [this PR](https://github.com/stevenleeg/geemusic/pull/54) or at my cloned repo. If you have any questions about setup, or recommendations for improving the setup directions, let me know and I will assist as quickly as possible.

**Comment by stevenleeg** [comment]
#54 is merged :+1: :tada:
